### PR TITLE
JIT llvm

### LINKED ISSUE #1621: LLVM Symbolic JIT issue
```
kafka@tubby:~/fun/tinygrad$ JIT=1 LLVM=1 LLVMOPT=1 python3 examples/llama.py --prompt="Hello." --temperature=0 --timing --count 20
using LLVM backend
using LLaMA-7B model
ram used: 13.48 GB, freqs_cis                                         : 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 292/292 [00:03<00:00, 80.81it/s]
loaded weights in 3623.43 ms, 13.48 GB loaded at 3.72 GB/s
Hello.
ran model in 2798.85 ms
sync in 213.23 ms
 I
ran model in 368.08 ms
Traceback (most recent call last):
  File "/Users/kafka/fun/tinygrad/examples/llama.py", line 446, in <module>
    logits = llama.model(Tensor([toks[start_pos:]]), start_pos).realize()[:, -1, :]
  File "/Users/kafka/fun/tinygrad/examples/llama.py", line 164, in __call__
    h = h.sequential([functools.partial(layer, start_pos=start_pos, freqs_cis=freqs_cis, mask=mask) for layer in self.layers])
  File "/Users/kafka/fun/tinygrad/tinygrad/tensor.py", line 680, in sequential
    def sequential(self, ll:List[Callable[[Tensor], Tensor]]): return reduce(lambda x,f: f(x), ll, self)
  File "/Users/kafka/fun/tinygrad/tinygrad/tensor.py", line 680, in <lambda>
    def sequential(self, ll:List[Callable[[Tensor], Tensor]]): return reduce(lambda x,f: f(x), ll, self)
  File "/Users/kafka/fun/tinygrad/examples/llama.py", line 130, in __call__
    output, cache_k, cache_v = self.jitted_attn(self.jitted_attention_norm(x), self.cache_k, self.cache_v, start_pos, freqs_cis, mask)
  File "/Users/kafka/fun/tinygrad/tinygrad/jit.py", line 23, in __call__
    if Device.DEFAULT not in JIT_SUPPORTED_DEVICE: return self.fxn(*args, **kwargs)  # only jit on supported device
  File "/Users/kafka/fun/tinygrad/examples/llama.py", line 88, in __call__
    keys, values = repeat_kv(keys, self.n_rep).realize(), repeat_kv(values, self.n_rep).realize()
  File "/Users/kafka/fun/tinygrad/tinygrad/tensor.py", line 94, in realize
    self.lazydata.realize()
  File "/Users/kafka/fun/tinygrad/tinygrad/lazy.py", line 158, in realize
    self.realized = Device[self.device].exec_ast(self.op, output=self, **self._device_extra_args())
  File "/Users/kafka/fun/tinygrad/tinygrad/ops.py", line 200, in exec_ast
    if k.key not in self.method_cache: self.method_cache[k.key] = get_program()
  File "/Users/kafka/fun/tinygrad/tinygrad/ops.py", line 197, in get_program
    return self.to_program(k)
  File "/Users/kafka/fun/tinygrad/tinygrad/ops.py", line 161, in to_program
    ret = self.renderer(k.function_name, k.uops)
  File "/Users/kafka/fun/tinygrad/tinygrad/renderer/llvmir.py", line 78, in uops_to_llvm_ir
    func_dtypes = [dtype_to_llvm_dtype[dtype] for dtype in buf_to_dtype.values()]
  File "/Users/kafka/fun/tinygrad/tinygrad/renderer/llvmir.py", line 78, in <listcomp>
    func_dtypes = [dtype_to_llvm_dtype[dtype] for dtype in buf_to_dtype.values()]
KeyError: dtypes._arg_int32
```

### PR DESCRIPTION
**Pull request by nimlgen**
Closes #1621

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/jit.py                            1               1
tinygrad/renderer/llvmir.py               18              14
tinygrad/runtime/ops_llvm.py               1               1
------------------------------------------------------------
total                                     20              16
------------------------------------------------------------
lines added in the tinygrad folder: 4
```

File: tinygrad/renderer/llvmir.py [hunk 1]
```diff
@@ -90,6 +91,9 @@ def uops_to_llvm_ir(function_name:str, uops:List[UOp]) -> Tuple[str, Optional[Li
   lvars: Dict[Optional[Token], Any] = {}  # this Any is an llvm type
   render_llvm[Variable] = lambda self,ops,ctx: lvars[self.expr]
 
+  for bufname,dtype in buf_to_dtype.items():
+    if dtype == dtypes._arg_int32: lvars[bufname] = bb[-1].sext(func.args[buf_index[bufname]], ir.IntType(64))
```
Comment (dc-dc-dc) [inline tinygrad/renderer/llvmir.py:95]: should this be `IntType(32)` since the dtype is a int32?

**Comment by wozeparrot** [comment]
conflicts

**Comment by nimlgen** [comment]
done

**Pull request merged**